src/sequence_state.cc

```diff
@@ -118,23 +151,43 @@ SequenceStates::Initialize(
       }
     }
 
-    std::shared_ptr<AllocatedMemory> data;
+    std::shared_ptr<MutableMemory> data;
     auto initial_state_it = initial_state.find(state_config.input_name());
     if (initial_state_it != initial_state.end()) {
-      data = std::make_shared<AllocatedMemory>(
-          initial_state_it->second.data_->TotalByteSize(),
-          TRITONSERVER_MEMORY_CPU, 0);
-
-      TRITONSERVER_MemoryType memory_type;
-      int64_t memory_type_id;
-      char* dst_buffer = data->MutableBuffer(&memory_type, &memory_type_id);
-      char* initial_state_buffer =
-          initial_state_it->second.data_->MutableBuffer(
-              &memory_type, &memory_type_id);
-
-      memcpy(
-          dst_buffer, initial_state_buffer,
-          initial_state_it->second.data_->TotalByteSize());
+      if (use_growable_memory) {
+        // TODO: use a constant or a flag for virtual address space size.
+        std::unique_ptr<GrowableMemory> growable_memory;
+        RETURN_IF_ERROR(GrowableMemory::Create(
+            initial_state_it->second.data_->TotalByteSize(),
+            TRITONSERVER_MEMORY_GPU, device_id, growable_memory,
+            1024 * 1024 * 1024));
```

Comment:
Can you add the constant and remove the TODO? `constexpr ... kDefaultVirtualGpuMemorySize = 1024*1024*1024;` or something like that?

Reply:
It is now a flag.